### PR TITLE
[Docs] [Branching] Chore/branching docs

### PR DESCRIPTION
**Pull request by MildTomato**
## What kind of change does this PR introduce?

Bug fix, feature, docs update, ...

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/platform/branching.mdx [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+  />
+  <figcaption>Each branch has a separate Supabase instance.</figcaption>
+</figure>
+
+Each time a commit is pushed with new migrations in the `./supabase/migrations` directory, the migrations are run on the matching preview instance.
+
+<figure className="max-w-[700px] mx-auto">
+  <Image
+    className="hidden dark:block"
+    alt="Each GitHub branch can have its own Supabase preview branch."
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:100]:
```suggestion
    alt="New migration files trigger migrations on the preview instance."
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+    src="/docs/img/guides/platform/branching/github-workflow-commit-migration.jpg"
+  />
+  <Image
+    className="dark:hidden"
+    alt="Each GitHub branch can have its own Supabase preview branch."
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:105]:
```suggestion
    alt="New migration files trigger migrations on the preview instance."
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+    src="/docs/img/guides/platform/branching/github-workflow-commit-migration--light.jpg"
+  />
+  <figcaption>Each GitHub branch can have its own Supabase preview branch.</figcaption>
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:108]:
```suggestion
  <figcaption>New migration files trigger migrations on the preview instance.</figcaption>
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+</figure>
+
+Only GitHub is currently supported (as a beta release). If you’re interested in other Git providers or a non-Git-based workflow, join the discussions for _GitLab Branching_, _BitBucket Branching_, and non-Git based Branching.
+
+### Alternatives to branching
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:113]:
```suggestion
## Alternatives to branching
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+
+   Create a single project to host your production instance. Test any changes locally, then run the migrations against your hosted production project.
+
+You can also combine both strategies to perform both local and staging tests.
+
+## How to use Supabase Branching
+
+<Admonition type="note">
+
+    Supabase branching requires the GitHub integration. Your application must be developed in a GitHub repository.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:133]:
```suggestion
Supabase branching requires the GitHub integration. Your application must be developed in a GitHub repository.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+
+Use the NextJS example template to try out branching. This template includes sample migration and seed files to get you started.
+
+Run the following command in your terminal to clone the example:
+
+```bash
+npx create-next-app -e with-supabase
+```
+
+Push your new project to a GitHub repo. For more information, see the GitHub guides on creating and pushing code to a new repository.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:156]:
```suggestion
Push your new project to a GitHub repo. For more information, see the GitHub guides on [creating](https://docs.github.com/en/get-started/quickstart/create-a-repo) and [pushing code](https://docs.github.com/en/get-started/using-git/pushing-commits-to-a-remote-repository) to a new repository.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 7]
```diff
@@ -0,0 +1,526 @@
...
+
+</Admonition>
+
+##### If your migration files are already in GitHub, under `./supabase/migrations`:
+
+If your migration files are up to date with your production database, you don’t need to do anything. You can now turn on branching.
+
+##### If you don’t have migration files in GitHub:
+
+You need to take a snapshot of your database. Log into Supabase from the command line client and pull the relevant files. Then commit them to GitHub:
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:176]:
```suggestion
You need to take a snapshot of your database. Log into Supabase from the [command line client](/docs/guides/cli) and pull the relevant files. Then commit them to GitHub:
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 8]
```diff
@@ -0,0 +1,526 @@
...
+
+1. Inside your project directory, initialize your configuration for Supabase local development:
+
+   ```bash
+   supabase init
+   ```
+
+2. Pull down database-related files. You can find your database URL
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:184]:
```suggestion
2. Pull down database-related files. You can find your database URL in your [database settings](https://supabase.com/dashboard/project/_/settings/database).
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 9]
```diff
@@ -0,0 +1,526 @@
...
+
+   ```bash
+   git add supabase
+   git commit -m "Initial migration"
+   git push
+   ```
+
+### Enable Supabase branching
+
+Once your repository is [correctly prepared](https://www.notion.so/Branching-docs-current-88961abecd9d437381a389b41e34ca7d?pvs=21), you can enable branching from the Supabase dashboard.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:200]:
```suggestion
Once your repository is [correctly prepared](#set-up-an-existing-project-for-branching), you can enable branching from the Supabase dashboard.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 10]
```diff
@@ -0,0 +1,526 @@
...
+
+<Admonition type="caution">
+
+Before turning on branching, make sure your [GitHub repository is prepared](https://www.notion.so/Branching-docs-current-88961abecd9d437381a389b41e34ca7d?pvs=21). If your repository doesn’t have all the migration files, your production branch could run an incomplete set of migrations.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:204]:
```suggestion
Before turning on branching, make sure your [GitHub repository is prepared](#set-up-an-existing-project-for-branching). If your repository doesn’t have all the migration files, your production branch could run an incomplete set of migrations.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+1.  Select the project you want to use, then click `Enable branching`.
+    If you’re prompted to join the waitlist, click `Join waitlist`. Access to branching will be expanded as branching graduates from alpha testing.
+
+    <figure className="max-w-[520px]">
+
+        <Image src="/docs/img/guides/platform/branching/enable-supabase-branching-opt-in-popover.jpg"/>
+
+    </figure>
+
+2.  If you see the following dialogue, you’re in the alpha testing group.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:224]:
```suggestion
2.  If you see the following dialog, you’re in the alpha testing group.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 12]
```diff
@@ -0,0 +1,526 @@
...
+
+    Return to your project and re-click `Enable branching`.
+
+3.  You should now see a popover with the GitHub Connection details shown.
+
+    Type in the branch you want to use for production. The name of the branch will be validated to make sure it exists in your GitHub repository.
+
+    <Admonition type="caution">
+
+        Your production branch can’t be changed while branching is enabled
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:258]:
```suggestion
        Your production branch can’t be changed while branching is enabled.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 13]
```diff
@@ -0,0 +1,526 @@
...
+
+    </Admonition>
+
+    <figure className="max-w-[520px]">
+
+        <Image src="/docs/img/guides/platform/branching/enable-supabase-branching-opt-in-install-production-branch.jpg"/>
+
+    </figure>
+
+4.  Click `I understand, enable branching`. Branching will then be enabled for your project.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:269]:
```suggestion
4.  Click `I understand, enable branching`. Branching is now enabled for your project.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 14]
```diff
@@ -0,0 +1,526 @@
...
+
+        <Image src="/docs/img/guides/platform/branching/enable-supabase-branching-first-preview-branch-branch-action.jpg" />
+
+    </figure>
+
+    Type in the branch name you want to add. Click `Create branch` to confirm.
+
+    <Admonition type="note">
+
+    ℹ️ **Only branches from the repository can be used to create a Preview Branch**
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:315]:
```suggestion
    **Only branches from the repository can be used to create a Preview Branch**
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 15]
```diff
@@ -0,0 +1,526 @@
...
+Supabase branching uses the state of your GitHub repository to apply migrations and seed preview instances. It checks the directory `supabase` at the top level of your repository.
+
+The `supabase` directory should include:
+
+- All SQL migration files, under the subdirectory `migrations`
+- An optional `seed.sql` file, used to seed preview instances with sample data
+
+The GitHub integration watches for changes in the `migrations` directory. When it detects a new migration file, it runs the new migration against the matching preview instance.
+
+You can create new migrations either [locally](https://www.notion.so/Branching-docs-current-88961abecd9d437381a389b41e34ca7d?pvs=21) or [remotely](https://www.notion.so/Branching-docs-current-88961abecd9d437381a389b41e34ca7d?pvs=21).
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:337]:
```suggestion
You can create new migrations either [locally](#develop-locally) or [remotely](#develop-remotely).
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 16]
```diff
@@ -0,0 +1,526 @@
...
+
+#### Develop remotely
+
+As an alternative to developing locally, you can make changes on your remote Supabase dashboard. You can then pull these changes to your local machine and commit them to GitHub.
+
+<Admonition type="note">
+
+Changes must be locally pulled and committed to keep your GitHub repository state in sync.
+
+Dashboard changes aren’t automatically reflected in your repository. If you’d like to see automatic syncing in a future release, join the discussion.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:390]: need link to discussion here

File: apps/docs/pages/guides/platform/branching.mdx [hunk 17]
```diff
@@ -0,0 +1,526 @@
...
+
+When you open a pull request on GitHub, the Supabase integration automatically checks for a matching preview branch. If one doesn’t exist, it gets created.
+
+A comment is added to your PR with the deployment status of your preview branch. Statuses are shown separately for Database, Services, and APIs.
+
+<figure className="max-w-[700px] mx-auto">
+  <Image src="/docs/img/guides/platform/branching/develop-your-app-open-pull-request-github.jpg" />
+  {/* prettier-ignore */}
+  <figcaption>
+    Supabase GitHub integration will comment on your PR the status of your Preview Branch, including wether migrations had run successfully.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:419]:
```suggestion
    Supabase GitHub integration will comment on your PR with the status of your Preview Branch, including whether migrations have successfully run.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 18]
```diff
@@ -0,0 +1,526 @@
...
+  </figcaption>
+</figure>
+
+Every time a new commit is pushed that changes the migration files in `./supabase/migrations`, the new migrations are run against the preview branch. You can check the status of these runs in the comment’s Tasks table.
+
+#### GitHub required check setting
+
+It is highly recommended turning on a ‘required check’ for the Supabase integration. You can do this from your GitHub repository settings.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:427]:
```suggestion
We highly recommend turning on a ‘required check’ for the Supabase integration. You can do this from your GitHub repository settings.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 19]
```diff
@@ -0,0 +1,526 @@
...
+
+This prevents PRs from being merged when migration checks fail, and stops invalid migrations from being merged into your production branch. (link to GitHub docs for this)
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:429]:
```suggestion
This prevents PRs from being merged when migration checks fail, and stops invalid migrations from being merged into your production branch.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 20]
```diff
@@ -0,0 +1,526 @@
...
+  <Image src="/docs/img/guides/platform/branching/github-required-check.jpg" />
+  {/* prettier-ignore */}
+  <figcaption>
+    Check the "Require status checks to pass before merging" option.
+  </figcaption>
+</figure>
+
+### Disable branching
+
+You can disable branching at any time. Navigate to the [Branches](https://supabase.green/dashboard/project/_/branches) page, which can be found via the Branches dropdown menu on the top navigation, click "Manage Branches" in the menu. Then click the 'Disable branching' button at the top of the Overview section.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:441]:
```suggestion
You can disable branching at any time. Navigate to the [Branches](https://supabase.green/dashboard/project/_/branches) page, which can be found via the Branches dropdown menu on the top navigation, then click "Manage Branches" in the menu. Click the 'Disable branching' button at the top of the Overview section.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 21]
```diff
@@ -0,0 +1,526 @@
...
+
+### Rolling back migrations
+
+In some circumstances, you might want to roll back changes you’ve made in an earlier migration change. For example, you might’ve pushed a migration file containing schema changes you no longer want.
+
+To fix this, push your latest changes, then delete the preview branch and reopen it.
+
+The new preview branch is reseeded from your `./supabase/seed.sql` file. Any additional data changes you made on the old preview branch are lost. This is equivalent to running `supabase db reset` locally. All migrations are rerun in sequential order.
+
+### Seeding behaviour
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:457]:
```suggestion
### Seeding behavior
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 22]
```diff
@@ -0,0 +1,526 @@
...
+
+With the Supabase branching integration, you can sync the Git branch used by the hosting provider with the corresponding Supabase preview branch. This means that the preview deployment built by your hosting provider is matched to the correct database schema, edge functions, and other Supabase configurations.
+
+### Vercel
+
+<Admonition type="note">
+
+Vercel Branching support is in development
+
+The Vercel Integration for Supabase branching is under development. To express your interest, join the discussion on GitHub discussions
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:497]: link here

File: apps/docs/pages/guides/platform/branching.mdx [hunk 23]
```diff
@@ -0,0 +1,526 @@
...
+
+</Admonition>
+
+Install the Vercel integration:
+
+- From the Vercel marketplace (link) or
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:503]: link here

File: apps/docs/pages/guides/platform/branching.mdx [hunk 24]
```diff
@@ -0,0 +1,526 @@
...
+- By clicking the blue `Deploy` button in a Supabase example app’s `README` file
+
+<Admonition type="note">
+
+For branching to work with Vercel, you also need the Vercel GitHub integration (link).
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:508]: link here

File: apps/docs/pages/guides/platform/branching.mdx [hunk 25]
```diff
@@ -0,0 +1,526 @@
...
+
+</Admonition>
+
+Supabase automatically updates your Vercel project with the correct environment variables for the corresponding preview branches.
+
+### Netlify
+
+A Netlify integration is under consideration. If you’re interested in branching with Netlify, join the GitHub discussion.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:516]: link here

File: apps/docs/pages/guides/platform/branching.mdx [hunk 26]
```diff
@@ -0,0 +1,526 @@
...
+
+## Feedback
+
+Supabase branching is a new and exciting new part of the Supabase development ecosystem. We’re monitoring its success and open to any feedback.
+
+You can join the conversation over in GitHub discussions.
```
Comment (charislam) [inline apps/docs/pages/guides/platform/branching.mdx:522]: link here

File: apps/docs/pages/guides/platform/branching.mdx (hunk 4 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:113]: Are you sure, feels like we're just removing the level 2 TOCs

File: apps/docs/pages/guides/platform/branching.mdx [hunk 27]
```diff
@@ -0,0 +1,502 @@
...
+1.  Select the project you want to use, then click `Enable branching`.
+    If you’re prompted to join the waitlist, click `Join waitlist`. Access to branching will be expanded as branching graduates from alpha testing.
+
+    <figure className="max-w-[700px]">
+
+        <Image src="/docs/img/guides/platform/branching/enable-supabase-branching-opt-in-popover.jpg"/>
+
+    </figure>
+
+2.  If you see the following dialogue, you’re in the alpha testing group.
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:224]:
```suggestion
2.  If you see the following dialog, you’re in the alpha testing group.
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 28]
```diff
@@ -0,0 +1,526 @@
...
+
+Supabase branches work similarly to Git branches. You can create a new branch for each set of related changes, such as new configurations, new database schemas, or new features. A temporary Supabase instance is created with your changes, so you can examine and test them without affecting your production instance.
+
+When you’re ready to ship your changes, merge your branch. Your production instance updates to reflect the new changes.
+
+<Admonition type="caution">
+
+It is not recommended to use Branching in a production environment
+
+Branching is an alpha feature. To try out branching for non-production experiments, request access here.
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:25]:
```suggestion
Branching is an alpha feature. To try out branching for non-production experiments, [request access here](https://forms.supabase.com/branching-request).
```

File: apps/docs/pages/guides/platform/branching.mdx [hunk 29]
```diff
@@ -0,0 +1,526 @@
...
+  />
+  <Image
+    className="dark:hidden"
+    alt="Each GitHub branch can have its own Supabase preview branch."
+    src="/docs/img/guides/platform/branching/github-workflow-commit-migration--light.jpg"
+  />
+  <figcaption>Each GitHub branch can have its own Supabase preview branch.</figcaption>
+</figure>
+
+Only GitHub is currently supported (as a beta release). If you’re interested in other Git providers or a non-Git-based workflow, join the discussions for _GitLab Branching_, _BitBucket Branching_, and non-Git based Branching.
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:109]:
```suggestion

Only GitHub is currently supported (as a beta release). If you’re interested in other Git providers or a non-Git-based workflow, join the discussions for [GitLab Branching, BitBucket Branching, and non-Git based Branching](https://github.com/orgs/supabase/discussions/18936).
```

File: apps/docs/pages/guides/platform/branching.mdx (hunk 26 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:520]:
```suggestion
You can join the [conversation over in GitHub discussions](https://github.com/orgs/supabase/discussions/18937).
```

File: apps/docs/pages/guides/platform/branching.mdx (hunk 25 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:514]:
```suggestion
A Netlify integration is under consideration. If you’re interested in branching with Netlify, [join the GitHub discussion](https://github.com/orgs/supabase/discussions/18938).
```

File: apps/docs/pages/guides/platform/branching.mdx (hunk 22 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:495]:
```suggestion
The Vercel Integration for Supabase branching is under development. To express your interest, [join the discussion on GitHub discussions](https://github.com/orgs/supabase/discussions/18938).
```

File: apps/docs/pages/guides/platform/branching.mdx (hunk 16 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:388]:
```suggestion

Dashboard changes aren’t automatically reflected in your repository. If you’d like to see automatic syncing in a future release, [join the discussion](https://github.com/orgs/supabase/discussions/18937).
```

File: apps/docs/pages/guides/platform/branching.mdx (hunk 24 above)
Comment (MildTomato) [reply apps/docs/pages/guides/platform/branching.mdx:507]:
```suggestion
For branching to work with Vercel, you also need the [Vercel GitHub integration](https://vercel.com/docs/deployments/git/vercel-for-github).
```

**Comment by MildTomato** [comment]
@charislam resolved all the issues I think

**Review by charislam — APPROVED** [review]

**Pull request merged**